bundles/framework/divmanazer/component/Chart.js

```diff
@@ -132,27 +137,41 @@ Oskari.clazz.define('Oskari.userinterface.component.Chart', function(sandbox, lo
         var padding = this.dimensions.label.padding;
         var lblCenterPadding = this.dimensions.label.verticalCenterPadding;
         //groups
-        var gy = this.svg.append("g")
-            .attr("class", "y axis")
-            .attr("transform", "translate(0," + (padding + lblCenterPadding) + ")")
-            .call(this.yAxis);
+        var gy = this.svg.append( "g" )
+            .attr( "class", "y axis" )
+            .attr( "transform", "translate(0," + (padding + lblCenterPadding) + ")" )
+            .call( this.yAxis );
 
-        var gx = this.svg.append("g")
-            .attr("class", "x axis")
-            .call(this.xAxis);
+        var gx = this.svg.append( "g" )
+            .attr( "class", "x axis" )
+            .call( this.xAxis );
+    },
+    /**
+     * @method setColorScale
+     */
+    setColorScale: function ( colors ) {
+        this.colorScale = d3.scaleQuantize()
+            .domain( [0, this.data.length] )
+            .range( colors );
```

Comment:
When used with thematic maps this is problematic. This component needs a method where the colorScale can be passed by the code using this component.